.github/workflows/modem__build-host-tests.yml

```diff
@@ -27,13 +27,19 @@ jobs:
             example: "simple_cmux_client"
             warning: "Warning: The smallest app partition is nearly full"
 
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     container: espressif/idf:${{ matrix.idf_ver }}
     steps:
-      - name: Checkout esp-protocols
+      - name: Check out code (v3)
```

Comment:
Why this is needed?

Reply:
In the older IDF releases, specifically versions earlier than 5.0, it was not possible to execute Github Action checkout@v4. This issue arose because it required Node 20, which, in turn, necessitated GLIBC_2.28. 
And, installing GLIBC_2.28 was incompatible with the older IDF releases (Ubuntu 18)

Reply:
could you please add this note to the comment, so we know why we cannot use `checkout@v4`